pkg/ccl/backupccl/restore_planning.go

```diff
@@ -392,12 +392,28 @@ func allocateDescriptorRewrites(
 					}
 					parentID = newParentID
 				}
-				// Check that the table name is _not_ in use.
+
+				// If we are restoring the table into an existing schema in the target
+				// database, we must ensure that the table name is _not_ in use.
 				// This would fail the CPut later anyway, but this yields a prettier error.
-				tableName := tree.NewUnqualifiedTableName(tree.Name(table.GetName()))
-				err := descs.CheckObjectNameCollision(ctx, col, txn.KV(), parentID, table.GetParentSchemaID(), tableName)
-				if err != nil {
-					return err
+				//
+				// If we are restoring the table into a schema that is also being
+				// restored then we do not need to check for collisions as the backing
+				// up cluster would enforce uniqueness of table names in a schema.
+				rw, ok := descriptorRewrites[table.GetParentSchemaID()]
+				restoringIntoExistingSchema := ok && rw.ToExisting
+				isSystemTable := table.GetParentID() == keys.SystemDatabaseID &&
+					table.GetParentSchemaID() == keys.SystemPublicSchemaID
```

Comment:
why check for both the `ParentID` and the `ParentSchemaID`?

Reply:
Unfortunately, the system database still has a pseudo public schema, i.e. the schema is not backed by a descriptor and so it will not end up `descriptorRewrites`. If the table we are restoring is part of the system database and part of this pseudo schema we want to check for collisions. If for some reason, a UDS ended up in the system database then that would be in `descriptorRewrites` and it would be treated like any other regular schema.